packages/trip-form/src/TripOptions/FeaturedOption.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import coreUtils from "@opentripplanner/core-utils";
+import { QuestionCircle } from "@styled-icons/fa-regular/QuestionCircle";
+
+import {
+  accessModeIsWalkOnly,
+  getCategoryModes,
+  getCategoryPrimaryMode,
+  getNonTransitModes,
+  getSelectedModes
+} from "./util";
+import { ModeOption, QueryProps } from "./types";
+import OptionButton from "./OptionButton";
+
+import * as S from "./styled";
+
+const FeaturedOption = ({
+  onQueryParamChange,
+  queryParams,
+  questionIcon,
+  setFeaturedOption,
+  supportedModes,
+  DetailedModeIcon
+}: QueryProps & {
+  setFeaturedOption(option: string): void;
+  questionIcon?: React.ReactElement;
+}): React.ReactElement => {
+  // No featured mode to show if walking to transit
+  if (accessModeIsWalkOnly(queryParams?.mode)) return null;
+  const nonTransitModes = getNonTransitModes(queryParams?.mode);
+  const selectedCompanies = queryParams?.company?.split(",") || [];
+  const selectedModes = getSelectedModes(queryParams);
+  const option = nonTransitModes[0];
+  const category = supportedModes.categories.find(c =>
+    getCategoryModes(c).some(o => o === option)
+  );
+  const optionsAreCheckboxes = Boolean(category.mode);
+
+  const selectOption = (isChecked: boolean, o: ModeOption) => {
+    let mode = selectedModes;
+    let company = selectedCompanies;
+    if (isChecked) {
+      // Un-check the company box if dealing with checkboxes. Otherwise, do nothing.
+      if (optionsAreCheckboxes) {
+        company = selectedCompanies.filter(c => c !== o.company);
+        // Do nothing if already radio button is already checked.
+      } else {
+        return;
+      }
+    } else {
+      // if un checked, set/add company and set mode (FIXME: what about car/walk)
+      if (o.mode) {
+        mode = selectedModes
+          .filter(coreUtils.itinerary.isTransit)
+          .concat([o.mode]);
+      }
+      if (o.company) {
+        company = optionsAreCheckboxes
+          ? selectedCompanies.concat([o.company])
+          : [o.company];
+      }
+    }
+    onQueryParamChange({
+      company: company.join(","),
+      mode: mode.join(",")
+    });
+  };
```

Comment:
The `onQueryParamChange` method needs to be invoked when this component mounts and when the mode is changed via the ModeRow, in addition to when a selection is made within the featured mode

As things are now the company value is only updated when the user interacts with the featured mode options so in the case of e-scooter when first selecting that mode the interface shows all the companies as being selected, but the query params say none of them are.  This is also needed to clear out old companies when switching the mode from bike rental to scooter rental, for example

As things are now the companies value isn't

Reply:
I've been trying to replicate this, but haven't been able to see this issue happening. When clicking any item in the mode row, I see the query params update correctly including company. However, I think the discrepancy might be due to the onQueryParamChange implementation for the story. Ours is as follows:

```js
  handleOnQueryParamChange = queryParam => {
    const { queryParams } = this.state;
    const newParams = { ...queryParams, ...queryParam };

    onQueryParamChange(queryParam);

    this.setState({
      queryParams: newParams
    });
  };
```

If you think this is the issue, we can bake this functionality built into the component.